### DISCUSSION TITLE
part-ii/2017-76/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#350] [Question comments]
Part II 2017 Graph Theory: Paper 2, Section II, H 

https://questions.tripos.org/part-ii/2017-76/

**Comment by stevezshen** [comment]
<details>
<summary>(b)</summary>

Let $i\sim j$ iff $a_{ij}>0$. Suppose that for some $A \subset [n]$, $\left|\Gamma(A)\right|<|A|$.  WLOG $A = \{1,2,\dots,j\}$ and $\Gamma(A) = \{1,2,\dots,j-1\}$, for some $j$. 

$a_{i,j-1} = 1 - \sum_{l=1}^{j-2} a_{i,j}$

$1 \geq \sum_{i=1}^{j} a_{i,j-1} = j - \sum_{i=1}^{j}\sum_{l-1}^{j-2}a_{i,l} = j - \sum_{l-1}^{j-2}\sum_{i=1}^{j}a_{i,l} \geq 2$ contradiction. 

So by Hall's theorem, exists a matching from $[n]$ to $[n]$, as required. 



</details>
